src/app/icd/client/DefaultICDClientStorage.cpp

```diff
@@ -439,7 +444,31 @@ CHIP_ERROR DefaultICDClientStorage::DeleteAllEntries(FabricIndex fabricIndex)
 
 CHIP_ERROR DefaultICDClientStorage::ProcessCheckInPayload(const ByteSpan & payload, ICDClientInfo & clientInfo)
 {
-    // TODO: Need to implement default decription code using CheckinMessage::ParseCheckinMessagePayload
+    uint32_t counter;
+    MutableByteSpan appData;
+    VerifyOrReturnError(chip::Protocols::SecureChannel::CheckinMessage::ParseCheckinMessagePayload(
+                            clientInfo.shared_key, payload, counter, appData) == CHIP_NO_ERROR,
+                        CHIP_ERROR_INVALID_ARGUMENT);
+    auto checkInCounter = (counter - clientInfo.start_icd_counter) % kCheckInCounterMax;
+
+    VerifyOrReturnError(checkInCounter > clientInfo.offset, CHIP_ERROR_INVALID_ARGUMENT);
+    clientInfo.offset = counter - clientInfo.start_icd_counter;
+    if (checkInCounter > kCheckInRolloverConstant)
+    {
+        RefreshKeyAndRegisterClient(clientInfo);
+    }
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR DefaultICDClientStorage::RefreshKeyAndRegisterClient(ICDClientInfo & clientInfo)
```

Comment:
We may refreshKey in separate PR and remove it from current PR.

Reply:
Offline : Decided to leave it upto the application